{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Preprocessing step\n",
    "\n",
    "The input data to run instaGRAAL can be generated by running **hicstuff**. This tool is available here: https://github.com/koszullab/hicstuff \n",
    "\n",
    "The Hi-C reads are iteratively mapped on the fragmented genome, then a sparse contact map is built.\n",
    "\n",
    "The whole pipeline can be run with the single command **hicstuff pipeline**. The options are:\n",
    "Here the parameters indicate: <br/>\n",
    "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--enzyme : the enzyme used to prepare the Hi-C library<br/>\n",
    "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--genome : the reference genome to map the reads against<br/>\n",
    "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--iterative : to do an iterative mapping<br/>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "hicstuff pipeline --enzyme DpnII \\\n",
    "    --genome GCF_000002985.6_WBcel235_genomic.fna \\\n",
    "    --iterative \\\n",
    "    -o instagraal_scaffolding \\\n",
    "    SRR3105481_1.fastq SRR3105481_2.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Running instaGRAAL\n",
    "\n",
    "## Scaffolding\n",
    "\n",
    "instaGRAAL is available at https://github.com/koszullab/instaGRAAL\n",
    "\n",
    "It can be run with the command **instagraal**. Among the options to run instaGRAAL:<br/>\n",
    "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--level : resolution of the contact map (default=4)<br/>\n",
    "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--cycles : number of iterations to run for each bin (default=100)<br/>\n",
    "\n",
    "All scaffoldings were run with 100 cycles. In the case of C.elegans --level=5, and in the case of D.melanogaster --level=4.\n",
    "The following parameters are, in order: the hicstuff output folder, the assembly to scaffold, the output folder name."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "instagraal -l 5 instagraal_scaffolding sim_c500/sim_c500__GCF_000002985.6_WBcel235_genomic.fa graal_l5"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The output scaffolds can be found in graal_l5/instagraal_scaffolding/test_mcmc_5/genome.fasta. Artifacts in the scaffolds can be corrected with the command **instagraal-polish**. This command uses the file info_frags.txt in the output folder, and the input assembly.\n",
    "\n",
    "## Polishing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "instagraal-polish -m polishing \\\n",
    "    -i graal_l5/instagraal_scaffolding/test_mcmc_5/info_frags.txt \\\n",
    "    -f sim_c500/sim_c500__GCF_000002985.6_WBcel235_genomic.fa \\\n",
    "    -o graal_l5/instagraal_scaffolding/test_mcmc_5/genome.polished.fasta"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
