{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The reference genome of Caenorhabditis elegans WBcel235 was retrieved from the NCBI database. This is a complete assembly, with 7 contigs : chr I, II, III, IV, V, X and mitochondrial.<br/>\n",
    "\n",
    "chr I&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;NC_003279.8&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;15072434<br/>\n",
    "chr II&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;NC_003280.10&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;15279421<br/>\n",
    "chr III&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;NC_003281.10&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;13783801<br/>\n",
    "chr IV&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;NC_003282.8&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;17493829<br/>\n",
    "chr V&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;NC_003283.11&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;20924180<br/>\n",
    "chr X&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;NC_003284.9&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;17718942<br/>\n",
    "MT&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;NC_001328.1&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;13794<br/>\n",
    "\n",
    "# Testing Hi-C data\n",
    "\n",
    "Hi-C data were downloaded from public databases. The data used here is registered under the reference SRR3105481, from the study \"A Streamlined Tethered Chromosome Conformation Capture Protocol\". The quality of the dataset was tested by mapping the Hi-C reads against the reference genome and building the contact map using the tool **hicstuff**.<br/>\n",
    "\n",
    "The full pipeline can be run with a single command line :"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "hicstuff pipeline --enzyme DpnII \\\n",
    "    --genome GCF_000002985.6_WBcel235_genomic.fna \\\n",
    "    --filter --iterative \\\n",
    "    -o hicstuff \\\n",
    "    SRR3105481_1.fastq SRR3105481_2.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here the parameters indicate: <br/>\n",
    "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--enzyme : the enzyme used to prepare the Hi-C library<br/>\n",
    "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--genome : the reference genome to map the reads against<br/>\n",
    "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--filter : to filter the reads<br/>\n",
    "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--iterative : to do an iterative mapping<br/>\n",
    "\n",
    "Then the contact map is built with the files generated by the pipeline :"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "hicstuff view --binning=100kb \\\n",
    "    --frags=fragments_list.txt \\\n",
    "    --normalize abs_fragments_contacts_weighted.txt \\\n",
    "    --cmap=RdYlGn"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The contact map represents the interaction frequencies in the whole genome. Six \"squares\" of heightened interaction frequencies are distinguisable: these are the six nuclear chromosomes. Intrachromosomal interactions are more frequent than interchromosomal interactions. The contact map can be used to evaluate the quality of an assembly.\n",
    "\n",
    "<img src=\"ref_100kb_RdYlGn.png\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Simulation of a fragmented assembly with different number of cuts\n",
    "\n",
    "The script fragment_genome.py makes a number of cuts, given as parameter -c, in the input genome. The simulated fragmented genome will then be used to test the scaffolders.<br/>\n",
    "\n",
    "Different number of cuts are tested, to generate more/less fragmented assemblies: 500, 1000, 2000.<br/>\n",
    "\n",
    "Here, the command makes 500 cuts in the reference genome :<br/>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "python3 fragment_genome.py --cut 500 --out sim_c500/sim_c500__GCF_000002985.6_WBcel235_genomic.fa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The output file used for scaffolding is called sim_c500_GCF_000002985.6_WBcel235_genomic.mixed.fa, where the fragments are organized in a random order.<br/>\n",
    "\n",
    "The statistics for the fragmented assemblies are as follow :<br/>\n",
    "\n",
    "## cut = 500\n",
    "\n",
    "stats for sim_c500/sim_c500_GCF_000002985.6_WBcel235_genomic.mixed.fa<br/>\n",
    "sum = 100286401, n = 507, ave = 197803.55, largest = 1066561<br/>\n",
    "N50 = 325954, n = 99<br/>\n",
    "N60 = 272956, n = 132<br/>\n",
    "N70 = 213945, n = 173<br/>\n",
    "N80 = 162591, n = 226<br/>\n",
    "N90 = 107197, n = 301<br/>\n",
    "N100 = 69, n = 507<br/>\n",
    "N_count = 0<br/>\n",
    "Gaps = 0<br/>\n",
    "\n",
    "## cut = 1000\n",
    "\n",
    "stats for sim_c1000/sim_c1000_GCF_000002985.6_WBcel235_genomic.mixed.fa<br/>\n",
    "sum = 100286401, n = 1007, ave = 99589.28, largest = 647248<br/>\n",
    "N50 = 170184, n = 183<br/>\n",
    "N60 = 137901, n = 249<br/>\n",
    "N70 = 105640, n = 331<br/>\n",
    "N80 = 80096, n = 440<br/>\n",
    "N90 = 55310, n = 588<br/>\n",
    "N100 = 156, n = 1007<br/>\n",
    "N_count = 0<br/>\n",
    "Gaps = 0<br/>\n",
    "\n",
    "## cut = 2000\n",
    "\n",
    "stats for sim_c2000/sim_c2000_GCF_000002985.6_WBcel235_genomic.mixed.fa<br/>\n",
    "sum = 100286401, n = 2007, ave = 49968.31, largest = 391496<br/>\n",
    "N50 = 81968, n = 383<br/>\n",
    "N60 = 69145, n = 516<br/>\n",
    "N70 = 54307, n = 681<br/>\n",
    "N80 = 41159, n = 893<br/>\n",
    "N90 = 26857, n = 1192<br/>\n",
    "N100 = 73, n = 2007<br/>\n",
    "N_count = 0<br/>\n",
    "Gaps = 0<br/>\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Building a contact map with the fragmented genome\n",
    "\n",
    "After simulating the fragmented assembly, the Hi-C reads can be mapped against it to build the contact map. The contact map should now look \"disorganised\".<br/>\n",
    "\n",
    "The previous steps to build the contact map with the reference genome are repeated with the fragmented assembly:<br/>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "hicstuff pipeline --enzyme DpnII \\\n",
    "    --genome GCF_000002985.6_WBcel235_genomic.fna \\\n",
    "    --filter --iterative \\\n",
    "    -o hicstuff \\\n",
    "    SRR3105481_1.fastq SRR3105481_2.fastq\n",
    "    \n",
    "hicstuff view --binning=100kb \\\n",
    "    --frags=fragments_list.txt \\\n",
    "    -c RdYlGn -o hicstuff \\\n",
    "    abs_fragments_contacts_weighted.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"sim_c500_100kb_hicstuff.png\">"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
