
# Preprocessing step

The input data to run instaGRAAL can be generated by running **hicstuff**. This tool is available here: https://github.com/koszullab/hicstuff 

The Hi-C reads are iteratively mapped on the fragmented genome, then a sparse contact map is built.

The whole pipeline can be run with the single command **hicstuff pipeline**. The options are:
Here the parameters indicate: <br/>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--enzyme : the enzyme used to prepare the Hi-C library<br/>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--genome : the reference genome to map the reads against<br/>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--iterative : to do an iterative mapping<br/>

# Running instaGRAAL

## Scaffolding

instaGRAAL is available at https://github.com/koszullab/instaGRAAL

It can be run with the command **instagraal**. Among the options to run instaGRAAL:<br/>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--level : resolution of the contact map (default=4)<br/>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--cycles : number of iterations to run for each bin (default=100)<br/>

All scaffoldings were run with 100 cycles. In the case of C.elegans --level=5, and in the case of D.melanogaster --level=4.
The following parameters are, in order: the hicstuff output folder, the assembly to scaffold, the output folder name.


```bash
instagraal -l 5 instagraal_scaffolding sim_c500/sim_c500__GCF_000002985.6_WBcel235_genomic.fa graal_l5
```

The output scaffolds can be found in graal_l5/instagraal_scaffolding/test_mcmc_5/genome.fasta. 

## Polishing

Artifacts in the scaffolds can be corrected with the command **instagraal-polish**. This command uses the file info_frags.txt in the output folder, and the input assembly.


```bash
instagraal-polish -m polishing \
    -i graal_l5/instagraal_scaffolding/test_mcmc_5/info_frags.txt \
    -f sim_c500/sim_c500__GCF_000002985.6_WBcel235_genomic.fa \
    -o graal_l5/instagraal_scaffolding/test_mcmc_5/genome.polished.fasta
```
