{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The quality of the scaffolds was evaluated with BUSCO and Quast, compared to the reference genome."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# BUSCO\n",
    "\n",
    "BUSCO is available at https://busco.ezlab.org/\n",
    "\n",
    "The lineages that were given as input to BUSCO were:\n",
    "- nematoda_odb9 for C. elegans\n",
    "- diptera_odb9 for D. melanogaster\n",
    "\n",
    "The parameters used to run BUSCO are: <br/>\n",
    "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-i : input sequence <br/>\n",
    "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-l : lineage retrieved from the BUSCO website <br/>\n",
    "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-m : mode, \"genome\" for genome assemblies <br/>\n",
    "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--long : optimization mode <br/>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "busco -i celegans_scaffolding.fa -o celegans_busco_output -l nematoda_odb9 -m genome --long\n",
    "busco -i dmelanogaster_scaffolding.fa -o dmelanogaster_busco_output -l diptera_odb9 -m genome --long"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "These commands were run for all the scaffoldings, the simulations and the reference genomes. Plots can then be generated by moving all the files containing \"short_summary\" in a folder, here called BUSCO_reports, and then using the command generate_plot.py. The only argument to give is -wd, which is the folder described previously."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "generate_plot.py -wd BUSCO_reports"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Caenorhabditis elegans\n",
    "\n",
    "<img src=\"busco_celegans.png\">\n",
    "<br/>\n",
    "**Figure 1. BUSCO assessment of C. elegans scaffoldings.** Although several replicates were run in the case of C. elegans, only the results for one replicate are displayed.<br/>\n",
    "<br/>\n",
    "<br/>\n",
    "\n",
    "## Drosophila melanogaster\n",
    "\n",
    "<img src=\"busco_dmelanogaster_sim_c2000.png\">\n",
    "<br/>\n",
    "**Figure 2. BUSCO assessment of D.melanogaster scaffoldings.** Simulation # cuts = 2000.<br/>\n",
    "<br/>\n",
    "<br/>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# QUAST\n",
    "\n",
    "QUAST is available here http://quast.sourceforge.net/quast\n",
    "\n",
    "QUAST was run with the reference genome and the reference genome's features.\n",
    "\n",
    "The arguments are: \n",
    "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-R : reference genome\n",
    "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--eukaryote : cause... well... it's an eukaryote\n",
    "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--features : genome features\n",
    "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--min-identity : minimum alignment identity\n",
    "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--labels : names to attribute to columns\n",
    "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--k-mer-stats : k-mer-based quality metrics \n",
    "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; list of assemblies to assess"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "quast.py -R reference.fa --eukaryote --features reference_features.gff -o sim_c500_quast --min-identity 99.9 \\\n",
    "    --labels \"3D-DNA, instaGRAAL, instaGRAAL-with-polish, SALSA2-with-correct, SALSA2-without-correct\" sim_c500/scaffolding* --k-mer-stats"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"dmelanogaster_sim_c2000_misassemblies_plot.png\">\n",
    "<br/>\n",
    "**Figure 3. Detection of misassemblies in D. melanogaster scaffolds.** Simulation # cuts = 2000.<br/>\n",
    "<br/>\n",
    "<br/>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
